src/input.rs

```diff
@@ -6,22 +6,82 @@ use content_inspector::{self, ContentType};
 
 use crate::error::*;
 
-const THEME_PREVIEW_FILE: &[u8] = include_bytes!("../assets/theme_preview.rs");
+/// A description of an Input source.
+/// This tells bat how to refer to the input.
+#[derive(Clone)]
+pub struct InputDescription {
+    pub(crate) name: String,
+
+    /// The input title.
+    /// This replaces the name if provided.
+    title: Option<String>,
+
+    /// The input kind.
+    kind: Option<String>,
+
+    /// A summary description of the input.
+    /// Defaults to "{kind} '{name}'"
+    summary: Option<String>,
+}
+
+impl InputDescription {
+    /// Creates a description for an input.
+    pub fn new(name: impl Into<String>) -> Self {
+        InputDescription {
+            name: name.into(),
+            title: None,
+            kind: None,
+            summary: None,
+        }
+    }
+
+    pub fn set_kind(&mut self, kind: Option<String>) -> () {
+        self.kind = kind;
+    }
+
+    pub fn set_summary(&mut self, summary: Option<String>) -> () {
+        self.summary = summary;
+    }
+
+    pub fn set_title(&mut self, title: Option<String>) -> () {
+        self.title = title;
+    }
 
-#[derive(Debug, Clone)]
-pub(crate) struct InputDescription {
-    pub full: String,
-    pub prefix: String,
-    pub name: String,
+    pub fn title(&self) -> &String {
+        match self.title.as_ref() {
+            Some(ref title) => title,
+            None => &self.name,
+        }
+    }
+
+    pub fn kind(&self) -> Option<&String> {
+        self.kind.as_ref()
+    }
+
+    pub fn summary(&self) -> String {
+        self.summary.clone().unwrap_or_else(|| match &self.kind {
+            None => self.name.clone(),
+            Some(kind) => format!("{} '{}'", kind.to_lowercase(), self.name),
```

Comment:
Why do we call `.to_lowercase` here? Wouldn't it be more flexible if we leave this out?

Reply:
Oh, because `summary` is only needed in the error message. Maybe we should rewrite that error message a bit and get rid of `.summary`? Is it needed anywhere else?

Reply:
It isn't needed elsewhere, but it might be useful to keep around for now in case we do use it.